src/presentation/test/integration/registration/generalPartner/address/postcode-general-partner-principal-office-address.test.ts

```diff
@@ -0,0 +1,95 @@
+import request from "supertest";
+
+import enTranslationText from "../../../../../../../locales/en/translations.json";
+import cyTranslationText from "../../../../../../../locales/cy/translations.json";
+
+import { appDevDependencies } from "../../../../../../config/dev-dependencies";
+import app from "../../../app";
+
+import { getUrl, setLocalesEnabled, toEscapedHtml, testTranslations } from "../../../../utils";
+import { POSTCODE_GENERAL_PARTNER_PRINCIPAL_OFFICE_ADDRESS_URL } from "presentation/controller/addressLookUp/url";
+import GeneralPartnerBuilder, {
+  generalPartnerLegalEntity
+} from "../../../../builder/GeneralPartnerBuilder";
+import AddressPageType from "../../../../../controller/addressLookUp/PageType";
+
+describe("Postcode general partner's principal office address page", () => {
+  const URL = getUrl(POSTCODE_GENERAL_PARTNER_PRINCIPAL_OFFICE_ADDRESS_URL);
+
+  beforeEach(() => {
+    setLocalesEnabled(false);
+    appDevDependencies.cacheRepository.feedCache(null);
+
```

Comment:
i would also initialise the generalPartnerGateway in teh beforeEach as some of the tests feed one in each time (it might be adding to the list of GP's each time)

use

`    appDevDependencies.generalPartnerGateway.feedGeneralPartners([]);
`

Reply:
For current tests it does not seem to be needed but I have added it as more tests will come when the other principal office address pages are added, so my be needed then.